cdap-ui/app/features/admin/controllers/namespace/templates-ctrl.js

```diff
@@ -59,7 +30,76 @@ angular.module(PKG.name + '.feature.admin')
 
           fetchConfig();
         });
-    });
+    }
+
+    if (!$stateParams.templateName) {
+      // Obtaining list of plugins
+      $scope.$watch(function () { return vm.pluginType; }, function () {
+        if (!vm.pluginType) { return; }
+
+        vm.plugin = null;
+        plugin = null;
+        vm.pluginConfig = null;
+
+        var prom;
+        switch (vm.pluginType) {
+          case 'source':
+            prom = myAdapterApi.fetchSources({ adapterType: vm.template }).$promise;
+            fetchApi = myAdapterApi.fetchSourceProperties;
```

Comment:
I feel the ```fetchApi``` should be passed on to the initialize function. It was a little hard to find its usage right away.